[Issue-24] - Jira metadata is gathered. Storing provided by unapproved pull request [Issue-8].

sgorbatiuk opened the pull request:
[Issue-24] - Jira metadata is gathered. Storing provided by unapproved pull request [Issue-8], so for now I wait for entry point for gathered Jira metadata (see line 76 in the `BuildVersionMetaDataCollector.java`).
Unfortunately, some tests still not implemented due to strange environment behavior—they passed in Eclipse but fail in the command line.

Commit:
Merge pull request #1 from webdizz/master

Pull changes from webdizz repo

sgorbatiuk committed:
Merge remote-tracking branch 'upstream/master'

sgorbatiuk committed:
Merge remote-tracking branch 'upstream/master'

sgorbatiuk committed:
Merge remote-tracking branch 'upstream/master'

sgorbatiuk committed:
Merge remote-tracking branch 'upstream/master'

sgorbatiuk committed:
Merge remote-tracking branch 'upstream/master'

sgorbatiuk committed:
Merge remote-tracking branch 'upstream/master'

sgorbatiuk committed:
Merge remote-tracking branch 'upstream/master'

sgorbatiuk committed:
Merge remote-tracking branch 'upstream/master'

sgorbatiuk committed:
[Issue-24] - Jira metadata is gathered. Storing provided by pull request
[Issue-8].

sgorbatiuk committed:
Fix dependencies.

webdizz commented on plugin/src/main/java/com/epam/jenkins/deployment/sphere/plugin/metadata/jira/JiraMetaDataCollector.java:
```diff
@@ -0,0 +1,164 @@
+    @Override
+    public Set<JiraIssueMetaData> collect(final AbstractBuild<?, ?> build, final TaskListener taskListener) {
+        // We can obtain JIRA connection details from JIRA Plugin only.
+        if (!isJiraPluginInstalled()) {
```
reverse, check for true then for false

webdizz commented on plugin/src/main/java/com/epam/jenkins/deployment/sphere/plugin/metadata/model/JiraIssueIdComparator.java:
```diff
@@ -0,0 +1,14 @@
+public class JiraIssueIdComparator implements Comparator<JiraIssueMetaData> {
+
+    @Override
+    public int compare(JiraIssueMetaData issue_1, JiraIssueMetaData issue_2) {
```
bad variable names

webdizz commented on plugin/src/test/java/com/epam/jenkins/deployment/sphere/plugin/metadata/collector/InvolvedBuildChangesCollectorTest.java:
```diff
@@ -0,0 +1,74 @@
+    @Ignore("Strange behavior: passed in Eclipse, but fails in console")
+    @Test
+    public void shouldReturnAnEmptyListWhenNoChangesWereMade() {
+//        Commented, because command line runs this test despite @Ignore 
```
do not check in commented code

sgorbatiuk committed:
Fixes according to remarks. Issues with tests for
InvolvedBuildChangesCollectorTest.java will be investigated, and tests
will be provided in the separate pull request.

ghost commented:
Fixes according to remarks. Issues with tests for `InvolvedBuildChangesCollectorTest.java` will be investigated, and tests will be provided in the separate pull request.
